app/controllers/api/migrations.php

```diff
@@ -0,0 +1,554 @@
+                'port' => $port,
+            ]),
+            'resources' => $resources,
+            'statusCounters' => '{}',
+            'resourceData' => "{}",
+            'errors' => []
+        ]));
+
+        $eventsInstance->setParam('migrationId', $migration->getId());
+
+        // Trigger Transfer
+        $event = new Migration();
+        $event
+            ->setMigration($migration)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($migration, Response::MODEL_MIGRATION);
+    });
+
+App::post('/v1/migrations/nhost/report')
+    ->groups(['api', 'migrations'])
+    ->desc('Generate a report on NHost Data')
+    ->label('scope', 'migrations.write')
+    // ->label('event', 'migrations.report')
+    // ->label('audits.event', 'migration.report')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'generateNHostReport')
+    ->label('sdk.description', '/docs/references/migrations/migration-nhost-report.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION_REPORT)
+    ->param('resources', [], new ArrayList(new WhiteList(NHost::getSupportedResources())), 'List of resources to migrate')
+    ->param('subdomain', '', new URL(), "Source's Subdomain")
+    ->param('region', '', new Text(512), "Source's Region")
+    ->param('adminSecret', '', new Text(512), "Source's Admin Secret")
+    ->param('database', '', new Text(512), "Source's Database Name")
+    ->param('username', '', new Text(512), "Source's Database Username")
+    ->param('password', '', new Text(512), "Source's Database Password")
+    ->param('port', 5432, new Integer(), "Source's Database Port", true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $subdomain, string $region, string $adminSecret, string $database, string $username, string $password, int $port, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        try {
+            $nhost = new NHost($subdomain, $region, $adminSecret, $database, $username, $password, $port);
+
+            $response
+                ->setStatusCode(Response::STATUS_CODE_CREATED)
+                ->dynamic(new Document($nhost->report($resources)), Response::MODEL_MIGRATION_REPORT);
+        } catch (\Exception $e) {
+            var_dump($e->getMessage());
+
+            $response
+                ->setStatusCode(Response::STATUS_CODE_BAD_REQUEST)
+                ->json([
+                    'status' => 'error',
+                    'message' => $e->getMessage()
+                ]);
```

Comment:
left overs and, this is not how we handle errors in the endpoint. Unless there is a very specific error we want to capture... 
else we just throw them and let the global handler do the rest

Reply:
Sorry, this was a fallback while I was trying to debug the issue with messages not being sent